tests/specs/lemmas.k

```diff
@@ -325,11 +336,34 @@ module LEMMAS-HASKELL [symbolic, kore]
     rule #range(M [ N := BUF:ByteArray ], L, K) => BUF [ L -Int N .. minInt(K, #sizeByteArray(BUF) -Int (L -Int N)) ] ++ #range(M, N +Int #sizeByteArray(BUF), K -Int minInt(K, #sizeByteArray(BUF) -Int (L -Int N)))  requires K  >Int 0 andBool L <Int N +Int #sizeByteArray(BUF) andBool L >=Int N                          [simplification]
     rule #range(M [ N := BUF:ByteArray ], L, K) => #range(M, L, K)                                                                                                                                                     requires K  >Int 0                                           andBool L >=Int N +Int #sizeByteArray(BUF) [simplification]
 
+
+    rule #range(M [ N := BUF:ByteArray ], N, K ) => 
+      BUF ++ #range( M, N +Int minInt(K, #sizeByteArray(BUF)), K -Int minInt (K, #sizeByteArray(BUF)))  
+      requires K >Int 0 [simplification]
+
+
```

Comment:
Counterexample: `#sizeByteArray(BUF) ==Int 2`, and `K ==Int 1`. This will incorrectly compute a bytearray that is too long (because it will take the entire `BUF`). Add this as a test, and fix the lemma, or remove the lemma and add any now-failing proofs to `tests/failing`.

Reply:
It looks like this rule:
```
    rule #range(M [ N := BUF:ByteArray ], L, K) => BUF [ L -Int N .. minInt(K, #sizeByteArray(BUF) -Int (L -Int N)) ] ++ #range(M, N +Int #sizeByteArray(BUF), K -Int minInt(K, #sizeByteArray(BUF) -Int (L -Int N)))  requires K  >Int 0 andBool L <Int N +Int #sizeByteArray(BUF) andBool L >=Int N                          [simplification]
```
is doing what I was trying to do with that rule, but doesn't work on the case `L ==Int N +Int #sizeByteArray(BUF)`. Is there any reason this lemma would not work with that condition relaxed to `L <=Int N +Int #sizeByteArray(BUF)`? It seems like in that case we get 
`#range( M [ N := BUF ], L, K ) => BUF [ L -Int N .. 0 ] ++ #range(M, L, K)` which is just `#range(M, L, K)`. Is this correct? If so, would it be OK if I modify that lemma and then remove my lemma?

Reply:
I removed the rule and updated the other rule to include the problematic case, and added tests for the updated rule.